pkg/asset/installconfig/aws/validation.go

```diff
@@ -349,10 +349,6 @@ var requiredServices = []string{
 
 // ValidateForProvisioning validates if the install config is valid for provisioning the cluster.
 func ValidateForProvisioning(session *session.Session, ic *types.InstallConfig, metadata *Metadata) error {
-	if ic.Publish == types.InternalPublishingStrategy {
```

Comment:
This will cause installs with internal publishing to fail when the user is not supplying a private hosted zone. Specifically, the later check for the public hosted zone should not be performed.

Reply:
The prior code just returned if not hosted zone was supplied. Is the correct logic here to return if no hosted zone was provided and using internal publishing?